### ISSUE TITLE
Create git hook  that converts aeon.ttl to aeon.owl using robot

### ISSUE THREAD

**Issue by andrecastro0o** [tibonto/aeon#72]
* requirements have [robot](http://robot.obolibrary.org/) install in local machine
* create file .git/hooks/pre-commit inside the aeon repository
* add the follow sh script to the .git/hooks/pre-commit
* once added a saved `chmod +x .git/hooks/pre-commit`

.git/hooks/pre-commit
```sh
#!/bin/sh
#
#  .git/hooks/pre-commit
#
robot convert --input aeon.ttl --format owl --output aeon.owl; git add aeon.owl

```
aeon.owl will be created and added to repository. The first time the hook runs it might require an extra commit to include the aeon.owl file

**Comment by andrecastro0o** [comment]
The shell script might only work on Unix systems. For Windows the `#!` might need to be modified to `#!C:/Program Files (x86)/Git/bin/sh.exe`
See https://www.gloomycorner.com/git-hooks-on-windows/ 
https://stackoverflow.com/questions/18277429/executing-git-hooks-on-windows

**Comment by StroemPhi** [comment]
@andrecastro0o Thank you for looking into this, will try to implement it asap.

**Comment by StroemPhi** [comment]
probably a dubplicate of #12

**Comment by StroemPhi** [comment]
> 
> 
> @StroemPhi The shell script might only work on Unix systems. For Windows the `#!` might need to be modified to `#!C:/Program Files (x86)/Git/bin/sh.exe`
> See https://www.gloomycorner.com/git-hooks-on-windows/
> https://stackoverflow.com/questions/18277429/executing-git-hooks-on-windows

Couldn't get this to run so far. Will try a .bat approch next.

**Comment by andrecastro0o** [comment]
the issue is not the git hook on windows, but how to run bash scripts on windows 8
See perhaps

https://stackoverflow.com/questions/6413377/is-there-a-way-to-run-bash-scripts-on-windows

**Comment by StroemPhi** [comment]
> 
> 
> the issue is not the git hook on windows, but how to run bash scripts on windows 8
> See perhaps
> 
> https://stackoverflow.com/questions/6413377/is-there-a-way-to-run-bash-scripts-on-windows

thank you for finding, will have a look at it tomorrow

**Comment by bencomp** [comment]
I'm not involved in any way and I know the original issue #12 was closed, but I can't help wondering: is there a reason to want to keep both a Turtle and RDF/XML version of the same data in version control?

**Comment by StroemPhi** [comment]
@bencomp the idea was to have the ttl for the editors (in a subfolder like src, very soon) and to have the owl at top level for the releases. So I huess the owl does not have to be produced with each single commit, but must be up to date with each release.

**Comment by bencomp** [comment]
@StroemPhi thanks for clarifying. Have you considered keeping one file as the source and using GitHub Actions and Releases to publish derivative files? That could prevent the need of keeping files in sync in version control.

**Comment by StroemPhi** [comment]
@bencomp hehe, funny that you mention this, as it has just been implemented since v.0.4.2 and [we presented this workflow improvement ](https://doi.org/10.5281/zenodo.5036013) last week at the VIVO conference.

I'm not really sure yet, if this is sufficient though. Looking at all the other OBO ontologies, an OWL is always part of the repo and usually the one in the main branch is used as the source for the PURL to resolve to. As we might want to submit AEON to OBO to get an OBO-PURL, this issue may become relevant for v.1 again. Thus I want to have it open until this is clearer.
